src/preamble.js

```diff
@@ -850,8 +851,10 @@ function demangle(func) {
   var hasLibcxxabi = !!Module['___cxa_demangle'];
   if (hasLibcxxabi) {
     try {
-      var buf = _malloc(func.length);
-      writeStringToMemory(func.substr(1), buf);
+      var s = func.substr(1);
+      var len = lengthBytesUTF8(s)+1;
```

Comment:
In locations like these where we know that the string is necessarily 7-bit ASCII in advance, we could optimize to use `string.length` instead of `lengthBytesUTF8`, and create a new `stringToAscii` serialization function which would be a tighter loop than `stringToUTF8` is.